tools/src/bin/docgen/main.rs

```diff
@@ -175,24 +179,36 @@ impl ItemMetadata {
             return Ok(ItemMetadata {
                 comment: Some(comment),
                 feature,
+                deprecation: None,
             });
         }
 
-        // If `prev` wasn't a comment, see if it was an expression_statement
-        // that itself was preceded by a comment.  This skips over
-        // expression-like preprocessor attributes on function decls.
-        if prev.kind() == "expression_statement" {
-            return match prev_prev {
-                Some(prev_prev) => Self::new(prev_prev, src),
-                None => Ok(ItemMetadata::default()),
+        // `prev` is a deprecation, `prev_prev` may be a comment, and `prev_prev_prev`
+        // may be a feature requirement.
+        if let Ok(deprecation) = Deprecation::new(prev, src) {
+            let comment = match prev_prev {
+                Some(prev_prev) => Comment::new(prev_prev, src).ok(),
+                None => None,
+            };
+            let prev_prev_prev = prev_prev.and_then(|prev_prev| prev_prev.prev_named_sibling());
```

Comment:
Consider making this part of the previous `match`?

Reply:
Done.